exporter/snowflakeexporter/exporter_traces.go

```diff
@@ -0,0 +1,274 @@
+		return fmt.Errorf("failed to create traces table: %w", err)
+	}
+
+	return nil
+}
+
+func (te *tracesExporter) shutdown(_ context.Context) error {
+	if te.db != nil {
+		return te.db.Close()
+	}
+	return nil
+}
+
+func (te *tracesExporter) tracesDataPusher(ctx context.Context, td ptrace.Traces) error {
+	te.logger.Debug("begin tracesDataPusher")
+	traceMaps := []map[string]any{}
+	for i := 0; i < td.ResourceSpans().Len(); i++ {
+		resourceSpan := td.ResourceSpans().At(i)
+
+		for j := 0; j < resourceSpan.ScopeSpans().Len(); j++ {
+			scopeSpan := resourceSpan.ScopeSpans().At(j)
+
+			for k := 0; k < scopeSpan.Spans().Len(); k++ {
+				span := scopeSpan.Spans().At(k)
+				eTimes, eNames, eAttributes, eDroppedCount := flattenEvents(span.Events(), te.logger)
+				lTraceIDs, lSpanIDs, lTraceStates, lAttributes, lDroppedCount := flattenLinks(span.Links(), te.logger)
+
+				traceMaps = append(traceMaps, map[string]any{
+					"rSchema":           resourceSpan.SchemaUrl(),
+					"rDroppedCount":     resourceSpan.Resource().DroppedAttributesCount(),
+					"rAttributes":       utility.ConvertAttributesToString(resourceSpan.Resource().Attributes(), te.logger),
+					"sSchema":           scopeSpan.SchemaUrl(),
+					"sName":             scopeSpan.Scope().Name(),
+					"sVersion":          scopeSpan.Scope().Version(),
+					"sDroppedCount":     scopeSpan.Scope().DroppedAttributesCount(),
+					"sAttributes":       utility.ConvertAttributesToString(scopeSpan.Scope().Attributes(), te.logger),
+					"traceID":           utility.TraceIDToHexOrEmptyString(span.TraceID()),
+					"spanID":            utility.SpanIDToHexOrEmptyString(span.SpanID()),
+					"traceState":        span.TraceState().AsRaw(),
+					"parentSpanID":      utility.SpanIDToHexOrEmptyString(span.ParentSpanID()),
+					"name":              span.Name(),
+					"kind":              span.Kind().String(),
+					"startTime":         span.StartTimestamp().AsTime(),
+					"endTime":           span.EndTimestamp().AsTime(),
+					"droppedCount":      span.DroppedAttributesCount(),
+					"attributes":        utility.ConvertAttributesToString(span.Attributes(), te.logger),
+					"statusMessage":     span.Status().Message(),
+					"statusCode":        span.Status().Code().String(),
+					"eventTimes":        eTimes,
+					"eventNames":        eNames,
+					"eventDroppedCount": eDroppedCount,
+					"eventAttributes":   eAttributes,
+					"linkTraceIDs":      lTraceIDs,
+					"linkSpanIDs":       lSpanIDs,
+					"linkTraceStates":   lTraceStates,
+					"linkDroppedCount":  lDroppedCount,
+					"linkAttributes":    lAttributes,
+				})
+			}
+		}
+	}
+
+	err := te.db.BatchInsert(ctx, traceMaps, te.insertSQL)
+	if err != nil {
+		return fmt.Errorf("failed to insert trace data: %w", err)
+	}
+	te.logger.Debug("end tracesDataPusher")
+	return nil
+}
+
+func flattenEvents(e ptrace.SpanEventSlice, l *zap.Logger) (pq.StringArray, pq.StringArray, pq.StringArray, pq.Int32Array) {
+	times := pq.StringArray{}
```

Comment:
Is PQ appropriate to use here? It's a postgres database driver, so it seems iffy to me to use it for snowflake.

Reply:
Snowflake seems to have a way to bind array values? Hard for me to tell if this works here.
https://pkg.go.dev/github.com/snowflakedb/gosnowflake#Array

Reply:
Yea I don't that feel great about it either. I'll check out that Array function and see how it works